lib/mongoid/extensions/boolean.rb

```diff
@@ -11,9 +11,14 @@ class << self
       # @example Mongoize the object.
       #   Boolean.mongoize("123.11")
       #
-      # @return [ String ] The object mongoized.
+      # @return [ true | false | nil ] The object mongoized or nil.
       def mongoize(object)
-        evolve(object)
```

Comment:
General comment which does not need to be fixed in this PR. Sometimes, `mongoize` calls `evolve`, and other times `evolve` calls `mongoize`. We should think about putting both methods side-by-side in the same `extensions` file, rather than also having separate `criteria/queryable/extensions` files.

Reply:
I'm not sure about moving the evolves, but we do have a ticket to standardize the functionality of evolve methods, so that should help with the idea that evolve behavior is different across types...

Reply:
Cool, problem should be looked at holistically (after solving `mongoize` first, of course!)